staging/src/k8s.io/client-go/applyconfigurations/core/v1/persistentvolumeclaim.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by applyconfiguration-gen. DO NOT EDIT.
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	v1 "k8s.io/client-go/applyconfigurations/meta/v1"
+)
+
+// PersistentVolumeClaimApplyConfiguration represents an declarative configuration of the PersistentVolumeClaim type for use
+// with apply.
+type PersistentVolumeClaimApplyConfiguration struct {
+	v1.TypeMetaApplyConfiguration   `json:",inline"`
+	v1.ObjectMetaApplyConfiguration `json:"metadata,omitempty"`
+	Spec                            *PersistentVolumeClaimSpecApplyConfiguration   `json:"spec,omitempty"`
+	Status                          *PersistentVolumeClaimStatusApplyConfiguration `json:"status,omitempty"`
+}
+
+// PersistentVolumeClaimApplyConfiguration constructs an declarative configuration of the PersistentVolumeClaim type for use with
+// apply.
+func PersistentVolumeClaim(name, namespace string) *PersistentVolumeClaimApplyConfiguration {
```

Comment:
Can I construct this from a PVC?

Reply:
Yes, via the "convert obj to apply object" functions that we haven't implemented yet ([KEP section](https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/2155-clientgo-apply#readmodifywrite-loop-support)). Once https://github.com/kubernetes-sigs/structured-merge-diff/pull/185 is merged I'll pick it up and generate the appropriate function.

Reply:
Yes if we're talking about constructing from extracting an existing object with the specific fieldmanager set.
No if it's about transforming a random non-apply object to an apply object, we should make that path as hard to implement as possible :-)